File: tools/resize.rb
require 'rmagick'
include Magick

path = '../public/images/works/'

exclude = ['.', '..', '.DS_Store', 'thumbs']

Dir.foreach(path) do |dir|
  next if exclude.include? dir
  
  dir += '/'

  Dir.foreach(path + dir) do |filename|
    next if exclude.include? filename
    
    image = ImageList.new(path + dir + filename).first

    image_resized = image.resize_to_fit(850, 650)
    image_resized.write (path + dir + filename)

    thumb = image.resize_to_fit(122, 122)
    thumb.write (path + dir + 'thumbs/' + filename)
  end
end

=begin
images = ImageList.new(file)

images.each do |image|
  thumb = img.scale(850, 650)
  thumb.write "thumb.jpg"
end
=end

File: tools/resize.ts
import * as fs from "fs";
import * as path from "path";
import sharp from "sharp";

const directoryPath = path.join(__dirname, "../public/images/works-original/");
const exclude = [".", "..", ".DS_Store"];

fs.readdir(directoryPath, async (err, directories) => {
  if (err) {
    console.error("Error reading directory:", err);
    return;
  }

  for (const dir of directories) {
    if (exclude.includes(dir)) {
      continue;
    }

    const sourcePath = path.join(directoryPath, dir);

    const files = await fs.promises.readdir(sourcePath);
    for (const filename of files) {
      if (exclude.includes(filename)) {
        continue;
      }

      const destPath = sourcePath.replace("/works-original/", "/works/");

      const imageSourcePath = path.join(sourcePath, filename);
      const imageDestPath = path.join(destPath, filename);
      const thumbPath = path.join(destPath, "thumbs", filename);

      fs.mkdirSync(path.dirname(thumbPath), { recursive: true });

      await Promise.all([
        sharp(imageSourcePath)
          .resize(850, 650, { fit: "inside" })
          .toFile(imageDestPath),
        sharp(imageSourcePath)
          .resize(122, 122, { fit: "cover" })
          .toFile(thumbPath),
      ]);
    }
  }
});
